spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/ratelimit/RedisRateLimiter.java

```diff
@@ -306,6 +319,30 @@ public Map<String, String> getHeaders(Config config, Long tokensLeft) {
 		return headers;
 	}
 
+	String getShard(int shards) {
+		if (shards > 0) {
+			// Ignore signature bit of shardIndex to make sure always positive value.
+			return String.valueOf((shardIndex.getAndIncrement() & Integer.MAX_VALUE) % shards);
+		}
+		return null;
+	}
+
+	int getShardedReplenishRate(Config config) {
+		int replenishRate = config.getReplenishRate();
+		if (config.getShards() > 0) {
+			replenishRate = replenishRate / config.getShards();
```

Comment:
if `replenishRate` is could not be divisible by shards, the actual rate will less than `replenishRate`, reminder value should be considered.